### PR TITLE
Use enum class instead of plain enum for PrimitiveType

### PR DESCRIPTION
**Pull request by johnBuffer**
Thanks a lot for making a contribution to SFML! 🙂

Before you create the pull request, we ask you to check the follow boxes. (For small changes not everything needs to ticked, but the more the better!)

* [ ] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before?
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [ ] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----

## Description

Use enum class instead of plain enum for PrimitiveType as it is recommended. However this breaks retrocompatibilty since `sf::VertexArray` will have to be declared using `PrimitiveType::Points` for example.

### CODE REVIEW

File: src/SFML/Graphics/RenderTarget.cpp [hunk 1]
```diff
@@ -706,7 +706,7 @@ void RenderTarget::drawPrimitives(PrimitiveType type, std::size_t firstVertex, s
     // Find the OpenGL primitive type
     static const GLenum modes[] = {GL_POINTS, GL_LINES, GL_LINE_STRIP, GL_TRIANGLES,
                                    GL_TRIANGLE_STRIP, GL_TRIANGLE_FAN, GL_QUADS};
-    GLenum mode = modes[type];
+    GLenum mode = modes[int(type)];
```
Comment (Bromeon) [inline src/SFML/Graphics/RenderTarget.cpp:709]: should probably be `static_cast<std::size_t>(type)`

**Review by Bromeon — COMMENTED** [review]
Why did you close it?

This would be a good change, but requires updating SFML to a newer standard. Since it breaks compatibility, it's only possible to have it for SFML 3.

[Upgrading SFML to C++17 is being planned.](https://github.com/SFML/SFML/wiki/SD%3A-CPP-17-Support)

**Comment by johnBuffer** [comment]
I closed it because the checks didn't pass thinking that reopening later when fixed would be better.

**Comment by eXpl0it3r** [comment]
Superseded by #2286

**Pull request closed without merging**